src/tri/_tri.cpp

```diff
@@ -650,24 +645,26 @@ PyObject* TriContourGenerator::contour_line_to_segs_and_kinds(const Contour& con
             contour_line.front() == contour_line.back())
             *(codes_ptr-1) = CLOSEPOLY;
 
-        if (PyList_SetItem(vertices_list, i, (PyObject*)segs) ||
-            PyList_SetItem(codes_list, i, (PyObject*)codes)) {
-            Py_XDECREF(segs);
-            Py_XDECREF(codes);
-            PyErr_SetString(PyExc_RuntimeError,
-                            "Unable to set contour segments and kind codes");
-            return NULL;
+        if (PyList_SetItem(vertices_list, i, segs.pyobj_steal()) ||
+            PyList_SetItem(codes_list, i, codes.pyobj_steal())) {
+            Py_XDECREF(vertices_list);
+            Py_XDECREF(codes_list);
```

Comment:
Note, I think these two Python lists may have leaked before, and I missed it in the previous PR. `segs` and `codes` in the new version won't leak since they are C++ objects with destructors.

Reply:
I think that because the list lengths are allocated earlier in the `PyList_New` call, the `PyList_SetItem` calls don't allocate any memory so will only raise an exception if the index is out of bounds.  But yes, it is good to have the `Py_XDECREF`s anyway.

Reply:
Maybe should use `PyList_SET_ITEM` which returns nothing, since we know it will not be out of bounds?

Reply:
Yes, good idea.